.github/workflows/CI.yml

```
@@ -0,0 +1,54 @@
+jobs:
+  build-and-analyze:
+    runs-on: ubuntu-latest  # Use ubuntu-latest for cross-platform compatibility
+
+    steps:
+      # 1. Checkout the repository
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      # 2. Setup .NET SDK
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v3
```

actionlint (1.7.4)

the runner of "actions/setup-dotnet@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)